oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+* Publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_.
+
+As noted above, please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time.
+
+Every published release should have a corresponding changelog entry.
+
+Application and Non-Library Repositories
+----------------------------------------
+
+When it comes to changelogs, the main difference between application and non-library repositories, and the eariler recommendations for library repositories, is that non-library code is not versioned.
+
+Since this code has no versions, the recommendations differ in the following ways:
+
+* The changelog entries will live in the CHANGELOG.rst file, following the recommendations detailed in `Changelogs and Changelog Entries`_.
+* Changelog entries should start with a date, rather than a date and version, because these repositories are not versioned.
+
+Changelog for edx-platform
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The edx-platform repository should have a CHANGELOG.rst, just like all other repositories.
+
+* Many notes that could potentially live in the edx-platform CHANGELOG.rst get added to the `Open Edx Release Planning`_ pages which have historically been created for each named release. Worst case, edx-platform notes are manually added to the CHANGELOG.rst instead, and at least the documentation stays more closely tied to the code.
+
+* Entries could be grouped by day if they could be added at a faster cadence (see `Tooling and Automation`_). Otherwise, date ranges should be used.
+
+* Attempting to add changelog entries at a very faster cadence without automation is likely to cause too many merge conflict issues. However, this should not stop people from adding entries immediately so we can learn.
```

Comment:
Is there anything we can do to help with merge conflicts? For the original edx-platform changelog, I had recommended not putting your entry at the very top of the file.  (Some developers then wanted to know what "near the top" meant, but maybe we've grown since then :)  )

Reply:
I don't think I've grown.  Do you mean if there are 5 bullet points for the current date's entry, you should insert your new bullet somewhere randomly between the first and last?  Or do you mean something different?

Either way, I pulled this code, but I am curious.

Reply:
Yes, I meant just choose a random spot that would still be appropriate.  At the time, we didn't demarcate the changelog by date (it was like edx-configuration's is now).